src/components/ErrorModal.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import Modal from 'react-bootstrap/Modal';
+import Button from 'react-bootstrap/Button';
+import { WsCloseCodeEnum } from '../enums/wsCloseCodeEnum';
+import { provider, roomID } from '../store/store';
+
+const ErrorModal: React.FC = () => {
+	const [infoModal, setInfoModal] = useState(false);
+	const [errorMessage, setErrorMessage] = useState<string | null>(null);
+	const [showLoginButton, setShowLoginButton] = useState(false);
+
+	const handleClose = () => {
+		setInfoModal(false);
+	};
+
+	const handleRedirect = () => {
+		if (window.location.host.startsWith('localhost')) {
```

Comment:
Repeated implementation should be refactored into a util-function and used in both files.